ReactiveCocoa/Swift/Signal.swift

```diff
@@ -724,16 +724,16 @@ extension SignalType {
 	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
 	public func skipRepeats(isRepeat: (Value, Value) -> Bool) -> Signal<Value, Error> {
 		return self
-			.map(Optional.init)
-			.combinePrevious(nil)
-			.filter { a, b in
-				if let a = a, b = b where isRepeat(a, b) {
-					return false
-				} else {
-					return true
+			.scan((nil, false)) { (accumulated: (Value?, Bool), next: Value) -> (Value?, Bool) in
+				switch accumulated.0 {
+				case .None:
+					return (next, false)
+				case let .Some(prev):
```

Comment:
I think this can be clearer if you separate both cases in separate `case`s:
```swift
case .None:
  return (next, false)
case let .Some(prev) where isRepeat(prev, next):
  return (prev, true)
default: 
  return (.Some(next), false)
}
```